src/vsr/README.md

```diff
@@ -1,3 +1,15 @@
+# Glossary
+
+- _checkpoint_: The process of moving items from the WAL to permanent storage. (Sidenote: in consensus literate this is often called snapshotting. But we use that term to mean something else.)
```

Comment:
(Apologies for the nitpicking!)
> - _checkpoint_: The process of moving items from the WAL to permanent storage. 

This is tricky. I think this is a little misleading as it is right now, but I'm not sure how to explain it succinctly enough for a glossary.
For one, the "checkpoint" itself doesn't actually remove/overwrite/move anything in the WAL — it doesn't touch the WAL at all. The "moving" (really copying) is handled by compaction, and then after the checkpoint, we can safely overwrite the WAL.
Also, s/"permanent storage"/grid, since we can re-write that data (via compaction) later.

s/"literate"/"literature"

Reply:
Good to be nit-picky!

What about

> _checkpoint_: The process whereby once entries in the WAL are compacted into the LSM tree, those entries are removed from the WAL. ( ... )

Reply:
We don't proactively "remove" anything from the WAL. Checkpointing just means that we are free to overwrite them (which we do as we wrap around the WAL).

How about:

_checkpoint_: Ensure that all updates from the past wrap of the WAL are durable in the grid, then advance the replica's recovery point by updating the superblock. After a checkpoint, the checkpointed WAL entries are safe to be overwritten by the next wrap.

Reply:
Cool that's fine! I realize `grid` makes more sense than saying `LSM tree` in here. I want to clarify grid though (I'm adding an entry). Grid is "the collection of all lsm trees and metadata about them", is that right?

Reply:
I would say that the grid is the section of the data file where the lsm trees and metadata reside. :+1:

Reply:
Awesome, added